packages/@azure/servicebus/data-plane/lib/session/messageSession.ts

```diff
@@ -920,6 +926,7 @@ export class MessageSession extends LinkEntity {
   private _deleteFromCache(): void {
     this._receiver = undefined;
     delete this._context.messageSessions[this.sessionId!];
+    delete this._context.expiredMessageSessions[this.sessionId!];
```

Comment:
When we get the error, first `_onAmqpError` is called, then `_onAmqpClose` is called. The latter will close the session receiver which will call this `_deleteFromCache` resulting in removal of the session from `expiredMessageSessions`.

Therefore, when we eventually call complete() on the message, the session is already removed from `expiredMessageSessions`

Reply:
yes, that's right - I fixed this in latest commit along with the tests